dask_jobqueue/tests/test_pbs.py

```diff
@@ -7,10 +6,80 @@
+        assert '#PBS -l select=1:ncpus=8:mem=27GB' in cluster.job_header
+        assert '#PBS -l walltime=00:02:00' in cluster.job_header
+        assert '#PBS -q' not in cluster.job_header
+        assert '#PBS -A' not in cluster.job_header
+
+    with PBSCluster(queue='regular', project='DaskOnPBS', processes=4, threads=2, memory='7GB',
+                    resource_spec='select=1:ncpus=24:mem=100GB') as cluster:
+
+        assert '#PBS -q regular' in cluster.job_header
+        assert '#PBS -N dask-worker' in cluster.job_header
+        assert '#PBS -l select=1:ncpus=24:mem=100GB' in cluster.job_header
+        assert '#PBS -l select=1:ncpus=8:mem=27GB' not in cluster.job_header
+        assert '#PBS -l walltime=' in cluster.job_header
+        assert '#PBS -A DaskOnPBS' in cluster.job_header
+
+    with PBSCluster() as cluster:
+
+        assert '#PBS -j oe' not in cluster.job_header
+        assert '#PBS -N' in cluster.job_header
+        assert '#PBS -l select=1:ncpus=' in cluster.job_header
+        assert '#PBS -l walltime=' in cluster.job_header
+        assert '#PBS -A' not in cluster.job_header
+        assert '#PBS -q' not in cluster.job_header
+
+    with PBSCluster(job_extra=['-j oe']) as cluster:
+
+        assert '#PBS -j oe' in cluster.job_header
+        assert '#PBS -N' in cluster.job_header
+        assert '#PBS -l select=1:ncpus=' in cluster.job_header
+        assert '#PBS -l walltime=' in cluster.job_header
+        assert '#PBS -A' not in cluster.job_header
+        assert '#PBS -q' not in cluster.job_header
+
+
+def test_job_script():
+    with PBSCluster(walltime='00:02:00', processes=4, threads=2, memory='7GB') as cluster:
+
+        job_script = cluster.job_script()
+        assert '#PBS' in job_script
+        assert '#PBS -N dask-worker' in job_script
+        assert '#PBS -l select=1:ncpus=8:mem=27GB' in job_script
+        assert '#PBS -l walltime=00:02:00' in job_script
+        assert '#PBS -q' not in job_script
+        assert '#PBS -A' not in job_script
+
+        assert '/dask-worker tcp://' in job_script
+        assert '--nthreads 2 --nprocs 4 --memory-limit 7GB' in job_script
+
+    with PBSCluster(queue='regular', project='DaskOnPBS', processes=4, threads=2, memory='7GB',
+                    resource_spec='select=1:ncpus=24:mem=100GB') as cluster:
+
+        job_script = cluster.job_script()
+        assert '#PBS -q regular' in job_script
+        assert '#PBS -N dask-worker' in job_script
+        assert '#PBS -l select=1:ncpus=24:mem=100GB' in job_script
+        assert '#PBS -l select=1:ncpus=8:mem=27GB' not in job_script
+        assert '#PBS -l walltime=' in job_script
+        assert '#PBS -A DaskOnPBS' in job_script
+
+        assert '/dask-worker tcp://' in job_script
+        assert '--nthreads 2 --nprocs 4 --memory-limit 7GB' in job_script
+
+
+#Run test only if py.test -E pbs is called, ie. when a docker compose PBS cluster is launched
+@pytest.mark.env("pbs")  # noqa: F811
```

Comment:
I think you can remove all the `noqa: F811` from all the files. They were for unused `loop` variables and I believe now `loop` is used everywhere.

By the way @mrocklin is adding `loop` to all the test functions a good general guideline (which what was done before)? I have to admit I don't understand this kind of subtleties yet.

Reply:
We do this in normal dask.distributed tests that use the normal/synchronous/blocking API.  The client is given this IOLoop explicitly and we clean it up after the test.  I honestly don't know if we're hooking everything together so that the IOLoops started by `*Cluster` are the same as the IOLoop that we hand in with the `loop` context manager.

We probably don't need to worry about this for now.  If at some point we run into odd conflicts or run out of file descriptors then this is something to come back to.

More information on common patterns when writing dask.distributed tests here: http://distributed.readthedocs.io/en/latest/develop.html#writing-tests

Reply:
From my tests and my understanding, the #noqa comment is still mandatory as `loop` is an import used as an arg name, which flake8 does not like much.

I did not used the loop parameters inside the basic test_header tests as they are not using a Tornado Loop. I understand from @lesteve comment and @mrocklin answer that it is the good choice.

Reply:
> From my tests and my understanding, the #noqa comment is still mandatory as loop is an import used as an arg name, which flake8 does not like much.

My bad I thought this was weird that the `noqa` was one the line on the `pytest.mark.env` and not on the line of the `def test_whatever(loop)` but it turns out this is how to make flake8 happy indeed.